morpheus/_lib/include/morpheus/stages/kafka_source.hpp

```diff
@@ -210,8 +229,9 @@ struct KafkaSourceStageInterfaceProxy
         std::map<std::string, std::string> config,
         bool disable_commit,
         bool disable_pre_filtering,
-        std::size_t stop_after = 0,
-        bool async_commits     = true);
+        std::size_t stop_after                           = 0,
+        bool async_commits                               = true,
+        std::optional<pybind11::function> oauth_callback = std::nullopt);
```

Comment:
Look at using the `PyFuncWrapper` from MRC to correctly decrement ref counts when this object is deallocated.